### ISSUE TITLE
Type Checking Fails

### ISSUE THREAD

**Issue by jlarmstrongiv** [withastro/starlight#2387]
### What version of `starlight` are you using?

0.28.2

### What version of `astro` are you using?

4.15.8

### What package manager are you using?

npm@10.8.3

### What operating system are you using?

Mac, Linux

### What browser are you using?

Node

### Describe the Bug

The last working version is 0.27.1

```
node_modules/@astrojs/starlight/utils/translations.ts(2,20): error TS2307: Cannot find module 'virtual:starlight/user-config' or its corresponding type declarations.
```

Same as the default starlight example.  Just run `npx astro check`

Confirmed by other users https://discord.com/channels/830184174198718474/1070481941863878697/1286068982134341692

It would be nice to have a test for with type checking for Starlight.  Most of the times upgrades with Starlight fail, it’s because of type checking.

### Link to Minimal Reproducible Example

https://stackblitz.com/github/withastro/starlight/tree/main/examples/basics?file=README.md&on=stackblitz

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by jlarmstrongiv** [comment]
One workaround is to:

Create a `starlight-virtual-internal.d.ts` file with the contents:

```ts
declare module "virtual:starlight/user-config" {
  var x: any;
  export = x;
}

declare module "virtual:starlight/plugin-translations" {
  var x: any;
  export = x;
}
```

And set `"skipLibCheck": true,` in your `tsconfig.json` file.

I found this bug causes type issues with newer versions of Astro and may cause issues without a `package-lock.json` file too.

**Comment by The-LukeZ** [comment]
This is still an issue

**Comment by delucis** [comment]
@The-LukeZ This will be fixed by #3572 which we’re planning to release this week.
